docs/notebooks/burger.ipynb

```diff
@@ -0,0 +1,899 @@
+    "                self._schedule.finalize(n1)\n",
+    "\n",
+    "        @action.register(Reverse)\n",
+    "        def action_reverse(cp_action):\n",
+    "            if self.reverse_step == 0:\n",
+    "                self.equation.adjoint_initial_condition()\n",
+    "            self.equation.adjoint(cp_action.n0, cp_action.n1, cp_action.clear_adj_deps, single_storage=self.single_storage)\n",
+    "            self.reverse_step += cp_action.n1 - cp_action.n0\n",
+    "            \n",
+    "        @action.register(Copy)\n",
+    "        def action_copy(cp_action):\n",
+    "            self.equation.copy_data(cp_action.n, cp_action.from_storage, cp_action.to_storage,\n",
+    "                                    move=False, single_storage=self.single_storage)\n",
+    "\n",
+    "        @action.register(Move)\n",
+    "        def action_move(cp_action):\n",
+    "            self.equation.copy_data(cp_action.n, cp_action.from_storage, cp_action.to_storage,\n",
+    "                                    move=True, single_storage=self.single_storage)\n",
+    "            \n",
+    "        @action.register(EndForward)\n",
+    "        def action_end_forward(cp_action):\n",
+    "            if self._schedule.max_n is None:\n",
+    "                self._schedule._max_n = self.max_n\n",
+    "            assert self.reverse_step == 0\n",
+    "            \n",
+    "        @action.register(EndReverse)\n",
+    "        def action_end_reverse(cp_action):\n",
+    "            if self._schedule.max_n != self.reverse_step:\n",
+    "                raise ValueError(\"The number of steps in the reverse phase\"\n",
+    "                                 \"is different from the number of steps in the\"\n",
+    "                                 \"forward phase.\")\n",
+    "            \n",
+    "        self.reverse_step = 0\n",
+    "        for _, cp_action in enumerate(self._schedule):\n",
+    "            action(cp_action)\n",
+    "            if isinstance(cp_action, EndReverse):\n",
+    "                break"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "## Adjoint-based sensitivity computations\n",
+    "\n",
+    "The purpose of this adjoint-based sensitivity computation is to use every checkpointing approach available in the `checkpoint_schedules` package and verify the consistent of the results. We employ a fundamental tool used in verification of gradients, which is the\n",
+    "*second order Taylor remainder convergence test*. Thus, let us consider the functional $I(u_0)$  and $\\nabla_{u_0}$ be its gradient with respect to the control parameter $u_0$. Let $u$ be the solution of the forward problem, and let $\\delta u$ be a perturbation to $u$. Therefore, the *Taylor remainder convergence test* is given by the expression: \n",
+    "$$ \\left|I(u + \\delta u) - I(u) - \\nabla_{u_0} I \\cdot \\delta u\\right| \\rightarrow 0 \\quad \\mathrm{at} \\ O(\\delta u^2).$$\n",
```

Comment:
The perturbation should be $\varepsilon \delta u$ with scalar $\varepsilon$ and direction $\delta u$, and then the Taylor remainder is $O(\varepsilon^2)$.